jwst/outlier_detection/utils.py

```diff
@@ -0,0 +1,208 @@
+                np.sum(badmask) / len(weight.flat) * 100))
+            badmasks.append(badmask)
+
+        # Fill resampled_sci array with nan's where mask values are True
+        for f1, f2 in zip(resampled_sci, badmasks):
+            for elem1, elem2 in zip(f1, f2):
+                elem1[elem2] = np.nan
+
+        del badmasks
+
+        # For a stack of images with "bad" data replaced with Nan
+        # use np.nanmedian to compute the median.
+        with warnings.catch_warnings():
+            warnings.filterwarnings(action="ignore",
+                                    message="All-NaN slice encountered",
+                                    category=RuntimeWarning)
+            median_image[row1:row2] = np.nanmedian(resampled_sci, axis=0)
+        del resampled_sci, resampled_weight
+
+    return median_image
+
+
+def compute_weight_threshold_container(resampled_models, maskpt):
+    '''
+    Compute weight means without keeping datamodels for each input open
+
+    Parameters
+    ----------
+    resampled_models : ~jwst.datamodels.ModelContainer
+        The input data models.
+
+    maskpt : float
+        The percentage of the mean weight to use as a threshold for masking.
+
+    Returns
+    -------
+    list
+        The weight thresholds for each integration.
+    '''
+
+    # Start by ensuring that the ModelContainer does NOT open and keep each datamodel
+    ropen_orig = resampled_models._return_open
+    resampled_models._return_open = True
+    # keep track of resulting computation for each input resampled datamodel
+    weight_thresholds = []
+    # For each model, compute the bad-pixel threshold from the weight arrays
+    for resampled in resampled_models:
+        weight = resampled.wht
+        weight_threshold = compute_weight_threshold(weight, maskpt)
+        weight_thresholds.append(weight_threshold)
+        # close and delete the model, just to explicitly try to keep the memory as clean as possible
+        resampled.close()
+        del resampled
+    # Reset ModelContainer attribute to original value
+    resampled_models._return_open = ropen_orig
+    return weight_thresholds
+
+
+def flag_crs_in_models(
+    input_models,
+    median_data,
+    median_wcs,
+    snr1,
+    snr2,
+    scale1,
+    scale2,
+    backg,
+    resample_data,
+):
+    for image in input_models:
+        if resample_data:
+            if 'SPECTRAL' not in image.meta.wcs.output_frame.axes_type:
+                input_pixflux_area = image.meta.photometry.pixelarea_steradians
+                # Set array shape, needed to compute image pixel area
+                image.meta.wcs.array_shape = image.shape
+                input_pixel_area = compute_image_pixel_area(image.meta.wcs)
+                pix_ratio = np.sqrt(input_pixflux_area / input_pixel_area)
+            else:
+                pix_ratio = 1.0
+
+            blot = gwcs_blot(median_data, median_wcs, image.data.shape, image.meta.wcs, pix_ratio)
+        else:
```

Comment:
Thanks for the question. This could certainly be cleaned up some more.

The call is equivalent. Within stcal [flag_resampled_model_crs](https://github.com/spacetelescope/stcal/pull/270/files#diff-a6ba118d5122b23fcaf8e8788d332274c632a441537646e4b63514d0402b9adeR194) begins with:
```python
    if not resample_data:
        return flag_crs(sci_data, sci_err, blot_data, snr1)
```
This could certainly be improved and was mostly carried over from the pre-refactor code which accepts a `resample_data` boolean:
https://github.com/spacetelescope/jwst/blob/859ac8ef69f01983308d8243b2143ae95895224b/jwst/outlier_detection/outlier_detection.py#L416-L417

I'll give this a look to see what would be involved to remove some of the `resample_data` args.